tools/workflow/cleaner/ecs/clean.go

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ecs
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+const (
+	Type          = "ecs"
+	taskDefPrefix = "taskdef"
+)
+
+var logger = log.New(os.Stdout, fmt.Sprintf("[%s] ", Type), log.LstdFlags)
+
+func Clean(sess *session.Session, expirationDate time.Time) error {
+	logger.Printf("Begin to clean ECS Task Definitions")
+	client := ecs.New(sess)
+
+	var nextToken *string
+	for {
+		ltdi := &ecs.ListTaskDefinitionsInput{NextToken: nextToken, MaxResults: aws.Int64(10)}
+		ltdo, err := client.ListTaskDefinitions(ltdi)
```

Comment:
Should we delete the services and task definition after de-registering it? In this way, we can delete the ECS too since the ECS does not have the creation time and we can use these two creation times to delete it.